extensions/community/JointConnector.json

```diff
@@ -0,0 +1,2029 @@
+{
+  "author": "",
+  "category": "General",
+  "extensionNamespace": "",
+  "fullName": "Joint Connector",
+  "helpPath": "",
+  "iconUrl": "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0ibWRpLWh1bWFuIiB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCI+PHBhdGggZD0iTTIxLDlIMTVWMjJIMTNWMTZIMTFWMjJIOVY5SDNWN0gyMU0xMiwyQTIsMiAwIDAsMSAxNCw0QTIsMiAwIDAsMSAxMiw2QzEwLjg5LDYgMTAsNS4xIDEwLDRDMTAsMi44OSAxMC44OSwyIDEyLDJaIiAvPjwvc3ZnPg==",
+  "name": "JointConnector",
+  "previewIconUrl": "https://asset-resources.gdevelop.io/public-resources/Icons/8834e0c63a962c74f1dab4c8f918c171261597341c4ca6574b300ab35855274b_human.svg",
+  "shortDescription": "Create and manage physics joints between two objects.",
+  "version": "1.0.0",
+  "description": [
+    "Weld joint",
+    "- Glue two objects together",
+    "- Joint is created at the center of the Joint Connector object",
+    "",
+    "Revolute joint",
+    "- Two objects are connected but can rotate on the joint",
+    "- Angles can be restricted and joints can have motors that spin the objects",
+    "- Joint is created at the center of the Joint Connector object",
+    "",
+    "Distance joint",
+    "- Two objects are forced to stay at set distance from each other",
+    "- Adjust the frequency property to change the stiffness of the spring",
+    "- Center of both objects must be inside the Joint Connector collision mask",
+    "",
+    "Rope joint",
+    "- Two objects can get closer to each other but they cannot exceed their starting distance",
+    "- Center of both objects must be inside the Joint Connector collision mask"
+  ],
+  "tags": [
+    "joint",
+    "ragdoll",
+    "physics",
+    "spring",
+    "rope",
+    "weld",
+    "revolute",
+    "distance"
+  ],
+  "authorIds": [
+    "IWykYNRvhCZBN3vEgKEbBPOR3Oc2",
+    "gqDaZjCfevOOxBYkK6zlhtZnXCg1",
+    "rotBq28wITdtfsrE7McHQri4k2w2"
+  ],
+  "dependencies": [],
+  "eventsFunctions": [
+    {
+      "description": "Connect overlapping physics objects with a revolute joint.",
+      "fullName": "Connect overlapping objects with a revolute joint",
+      "functionType": "Action",
+      "name": "CreateRevoluteJoints",
+      "sentence": "Connect overlapping _PARAM1_ with a revolute joint located at the center of _PARAM3_",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "const getProperty = function (propertyName) {",
+            "    return this._behaviorData[propertyName];",
+            "}",
+            "",
+            "gdjs.behaviorsTypes.items['JointConnector::RevoluteJoint'].prototype.getProperty = getProperty;",
```

Comment:
This is not a very good idea...

- This changes the protoype which is not great optimisation-wise
- this._behaviorData would not exist anymore if we were to mark it as private in the source file, as the minifier would be able to minify it if it knows nothing outside should be able to access it (and it should not be)
- Generally, this accesses an _ marked property (`_behaviorData`), which is completely disallowed by the extensions best practices (not, adding an indirection through a function call on the prototype does not make this magically OK)

Instead, you can use `jointBehavior._get<PropertyName>()`, e.g. `jointBehavior._getMotorEnabled()`. Those are the only exceptions to the `_` marked variables, as they are marked as such to tell only _users of the extension & creators of other extensions_ not to use it, but the extension that makes the behavior is still allowed to access it, as the naming convention is otherwise stable and guaranteed.

Reply:
Thank you very much for helping me; I am truly out of my depth at this point.  Here is what I tried (it seems to work until the last line):
```
const behaviorName = eventsFunctionContext.getBehaviorName("Physics");
const jointObjects = eventsFunctionContext.getObjects("Joint");
const linkManager = gdjs.LinksManager.getManager(runtimeScene);

for (const jointObject of jointObjects) {
    // Save properties of each joint
    const jointBehavior = eventsFunctionContext.getBehaviorName("RevoluteJoint");
   // The line below causes "not a function" error
    const enableLimit = jointBehavior._getPropertyAngleLimitsEnabled();
```
Error:
```
Uncaught TypeError: jointBehavior._getPropertyAngleLimitsEnabled is not a function
```

Can you provide an example of the *right* way to do this?  Then I will be able to replicate it on all functions of this extension.

Reply:
That was almost it, you only needed to remove the `Property` part. I went ahead and pushed the correction for you. I also auto-formatted the white-spaces (Shift-Alt-F in the code editor 😉) and removed a useless and likely forgotten console.log.